openedx_tagging/core/tagging/api.py

```diff
@@ -107,3 +107,73 @@ def tag_object(
     """
 
     return taxonomy.tag_object(tags, object_id, object_type)
+
+
+def autocomplete_tags(taxonomy: Taxonomy, prefix: str, object_id: str= None, count=10) -> List[TagResult]:
+    """
+    Returns the first `count` tag values in the given Taxonomy with names 
+    that begin with the given prefix string. The result is returned in alphabetical
+
+    Closed taxonomies return tag values that exist in the taxonomy. Also excludes all Tags that the
+    `object_id` already has. 
+
+    Free-text taxonomies return only tag values that are currently in use on (another) object. Also excludes
+    the tags used by `object_id`.
+    """
+    result = []
+
+    # Fetch tags that the object already has to exclude them from the result
+    excluded_tags = ObjectTag.objects.filter(object_id=object_id, taxonomy=taxonomy)
```

Comment:
Please add the appropriate index for this query.

Reply:
Sure, [added](https://github.com/openedx/openedx-learning/pull/59/files#diff-204875f41064da3abfd4648611ad41c6f21abe09277263410e3daafbd343f2a3R352) :+1: